banzai_nres/tests/test_traces.py

```diff
@@ -181,3 +181,26 @@ def make_simple_traces(nx=401, ny=403, trace_half_width=6, blaze=True):
         test_data += gaussian(y2d, trace_centers[i], 3, a=10000.0) * blaze_function
         input_traces[np.abs(y2d - trace_centers[i]) <= trace_half_width] = i + 1
     return test_data, trace_centers, input_traces
+
+
+def test_refine_traces_curved_trace():
+    nx, ny = 1001, 1003
+    expected_trace = np.zeros((ny, nx), dtype=int)
+
+    x0 = 498
+    input_y_center = 2e-4 * (np.arange(nx) - x0) ** 2.0 + 0.01 * (np.arange(nx) - x0) + 502.0
+    trace = np.round(input_y_center).astype(int)
+    for i in np.arange(nx, dtype=int):
+        for j in range(-7, 8):
```

Comment:
The trace width here is hard-coded to the same width as is currently used TRACE_HALF_HEIGHT in traces.py. It might be desirable (thought not necessarily straightforward) to just use TRACE_HALF_HEIGHT here, in case we want to change that value later. 

You might also consider iterating using a named variable in order to make it more clear as to why we are iterating over (-7, 8).

Reply:
I agree, it is cleaner to have trace width be a variable and not hard coded anywhere. I second both of the above comments.